features/FxA-16-devices-view/README.md

```diff
@@ -4,5 +4,174 @@ Device View - Basic UI
+Stories
+-------
 
+As a FF user when using Firefox,
+I want to know the devices that are connected to my Firefox Account,
+and important information about the status of those devices.
+
+I want my FF devices to be logically named
+so I recognize them and have the ability to rename them.
+I want to easily see the type and browser details of the each device,
+and to be able to identify the device I am currently using.
+
+I want a place to go when I think that something might be wrong with Sync,
+to see when each device last checked in with the Mozilla servers.
+
+I want a place to go when I think that my account
+may have been used without my permission,
+to check what devices are connected
+and to disconnect devices that I do not recognize.
+
+
+UI Mocks
+--------
+
+@ryanfeeley's mocks go here
+
+
+Details
+-------
+
+#### What is a "device"?
+
+By "device" here we mean "an instance of Firefox".
+There is no separate notion of having multiple "browsers" on a single "device".
+Technically the concept is closer to "profiles"
+but this is not a good name to present the users.
+
+* When I use a desktop machine
+  and connect both Firefox Release and Firefox Nightly to my account,
+  they appear as two separate devices.
+  There will be no indicatiion that they're on the same physical machine
+  apart from similarity in their auto-generated name and their UA details.
+
+* When I use Firefox for Android or Firefox for iOS...
+  TODO what exactly do we expect to happen?
+
+* When I use a FirefoxOS device,
+  it appears in the view as a single device
+  since the connetion is managed at the system level.
+  Also, TODO do we care at this stage?
+
+For the initial version, only things that connect to sync count as devices.
+
+* When I log in to Marketplace with my Firefox Account,
+  using the web-based flow in a browser that is not connected to my account,
+  the resulting session does *not* appear in the list of devices.
+
+* When I log in to Pocket with my Firefox Account
+  using non-Firefox web browser,
+  the resulting session does *not* appear in the list of devices.
+
+* When I have existing Firefox instances
+  that are already connected to sync,
+  they will appear in the devices view automatically.
+
+* TODO this also means that FirefoxOS should not be in the list
+  because it doesn't connect to sync yet.
+  It will soon though and it's something for us to think about.
+  
+
+
+#### What details do we show about the device?
+
+For each device we will show:
+
+* The human-readable name for the device,
+  either an auto-generated one or one provided by the user.
+
+* The time it was last seen,
+  i.e. the times at which it last used its session token
+  to interact with the FxA servers.
```

Comment:
FTR, I think we could track last-sync time by doing something like:

* Include an "fxa-device" field in the "idpClaims" section of signed assertions, alongside our existing fxa extension data
* Changing sync's tokenserver to extract this field and keep track of the last time each device requested a new sync token
* Either reflecting that back to the accounts service, or having content-server ask sync for it directly

It adds another system to the mix, and raises some architectural issues w.r.t cross-system information flow and ownership.  I think this will be the right thing to do *at some point*, my question is whether to block the rest of the devices view MVP on doing it.

(We could also have clients report their last-sync time explicitly, but that has the disadvantage of missing out on older clients, and adds work for e.g. the Fennec team).